src/NUnitTestAdapter/AdditionalTestInfo.cs

```diff
@@ -0,0 +1,58 @@
+﻿// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole, Terje Sandstrom
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System.Collections.Generic;
+using System.Xml;
+
+namespace NUnit.VisualStudio.TestAdapter
+{
+    internal class AdditionalTestInfo : IAdditionalTestInfo
+    {
+        private readonly Dictionary<TestKey, List<XmlNode>> _nodes = new Dictionary<TestKey, List<XmlNode>>();
```

Comment:
I'm not a fan of keeping data out of band like this. The name of this class is very general and I'm afraid it will add a concept that makes the adapter harder to understand.

Can you move the dictionary into the listener, since the listener is the one responsible for the decision to buffer instead of sending the output messages immediately?

Reply:
done